nginx.conf

```diff
@@ -52,13 +52,20 @@ http {
       proxy_pass http://localhost:8383;
       proxy_redirect off;
 
-      add_header Set-Cookie $session_cookie;
-      proxy_set_header X-Forwarded-Proto https;
-
       # buffer requests, but not responses, so streaming out works.
       proxy_request_buffering on;
       proxy_buffering off;
       proxy_read_timeout 2m;
+
+      # Dev-specific hacks:
+
+      # Remap "Set-Cookie: __Host-session=..." to "Set-Cookie: session=..."
+      # TODO but why is this helpful?
+      add_header Set-Cookie $session_cookie;
```

Comment:
The browser will only use a cookie prefixed with `__Host-` if it meets certain conditions, including that the response is from an HTTPS origin. Backend doesn't use HTTPS in local development, but we still want to use the cookie that it returns, so we (1) strip `__Host-` from the cookie, saving the result in `$session_cookie`, then (2) use `$session_cookie` for the cookie header when sending the next request. `map` on L24 above implements (1), then `add_header` here implements (2). I think that `map` can't be in `server`, which is why (1) and (2) are so far apart. This strategy works because Backend looks for a cookie whose name ends with `session`, not specifically for `__Host-session`: https://github.com/getodk/central-backend/blob/5aa8ff4a4e91a65574b19774e655a54c36999209/lib/http/preprocessors.js#L103

Reply:
Interesting!  That cookie parsing looks risky...

Reply:
@matthew-white i've updated the cookie comments :slightly_smiling_face: